hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/StreamWriteOperatorCoordinator.java

```diff
@@ -396,7 +396,11 @@ private void initInstant(String instant) {
         reset();
       } else {
         LOG.info("Recommit instant {}", instant);
-        commitInstant(instant);
+        boolean success = commitInstant(instant);
+        if (success) {
+          LOG.info("instant {} ReSync Hive", instant);
+          syncHive();
+        }
```

Comment:
Do we need to sync hive on start up ? The subsequent checkpoints will sync hive anyway.

Reply:
Hi Danny, want to bring up another topic here. If the commit is not success, e.g., the last batch has no data, we will reuse the instant of the last batch, and in the meanwhile, we will also start a new instant. This then leads to an inconsistent ckp_meta.

We encounter this issue this our inner branch. I could fire up a fix for this if necessary.